src/librustdoc/clean/types.rs

```diff
@@ -1966,11 +1967,58 @@ crate struct Static {
 }
 
 #[derive(Clone, PartialEq, Eq, Hash, Debug)]
-crate struct Constant {
-    crate type_: Type,
-    crate expr: String,
-    crate value: Option<String>,
-    crate is_literal: bool,
+crate enum Constant {
+    TyConst { type_: Type, expr: String },
+    Generic { type_: Type, body: BodyId },
+    Inline { type_: Type, did: DefId },
+    Const { type_: Type, did: DefId, body: BodyId },
+}
```

Comment:
Can you document what these are? I can guess `Inline` is for `const { ... }` blocks, but I can't tell about the rest.

Reply:
Arf sorry, I was experimenting on it for so long that I didn't document things...